airflow/utils/context.py

```diff
@@ -161,54 +178,89 @@ class OutletEventAccessor:
     :meta private:
     """
 
-    raw_key: str | Asset | AssetAlias
+    key: str | BaseAssetUniqueKey
     extra: dict[str, Any] = attrs.Factory(dict)
     asset_alias_events: list[AssetAliasEvent] = attrs.field(factory=list)
 
-    def add(self, asset: Asset | str, extra: dict[str, Any] | None = None) -> None:
+    def add(self, asset: str | Asset, extra: dict[str, Any] | None = None) -> None:
```

Comment:
Not sure if we want to support str here, based on the same reasoning. Maybe this should simply accept only Asset. (We can add AssetRef support later.)

Reply:
The main reason for keeping it is https://github.com/apache/airflow/issues/43956#issuecomment-2511443068. It supposed to be removed eventually

Reply:
I think we already have a deprecation warning for this one https://github.com/apache/airflow/blob/99e713efa398bef9d76eb4e8145538e828a05720/airflow/utils/context.py#L180

could I just remove all these string handling?

Reply:
I think we can just remove them.

Reply:
removed